api/src/main/java/org/apache/iceberg/catalog/Catalog.java

```diff
@@ -26,12 +26,22 @@
 import org.apache.iceberg.Transaction;
 import org.apache.iceberg.exceptions.AlreadyExistsException;
 import org.apache.iceberg.exceptions.NoSuchTableException;
+import org.apache.iceberg.exceptions.NotFoundException;
 
 /**
  * A Catalog API for table create, drop, and load operations.
  */
 public interface Catalog {
 
+  /**
+   * Return all the identifiers under this namespace.
+   *
+   * @param namespace a namespace
+   * @return an array of identifiers for tables
+   * @throws  NotFoundException if the namespace is not found
+   */
+  TableIdentifier[] listTables(Namespace namespace);
```

Comment:
Why return identifiers as an array instead of using a `List`? The reason why it's an array in Spark is because it's difficult to share collections between Scala and Java.

Reply:
I don't have a strong preference on either, let me change to use `List`.